Extensions/TweenBehavior/tweenruntimebehavior.js

```diff
@@ -0,0 +1,640 @@
+    this._removeTween(identifier)
+  }
+  return this.owner.deleteFromScene(this.runtimeScene);
+};
+
+gdjs.TweenRuntimeBehavior.prototype._removeTween = function(identifier) {
+  this._tweens[identifier].instance.dispose();
+  delete this._tweens[identifier];
+};
+
+/**
+ * Add an object variable tween.
+ * @param {string} identifier Unique id to idenfify the tween
+ * @param {objectvar} variable The object variable to store the tweened value
+ * @param {number} fromValue Start value
+ * @param {number} toVaue End value
+ * @param {string} easingValue Type of easing
+ * @param {number} durationValue Duration in milliseconds
+ * @param {boolean} destroyObjectWhenFinished Destroy this object when the tween ends
+ */
+gdjs.TweenRuntimeBehavior.prototype.addObjectVariableTween = function (
+  identifier,
+  variable,
+  fromValue,
+  toValue,
+  easingValue,
+  durationValue,
+  destroyObjectWhenFinished
+) {
+
+  if (this._tweenExists(identifier)) {
+    this.removeTween(identifier);
+  }
+
+  var newTweenable = new shifty.Tweenable();
+
+  newTweenable.setConfig({
+    from: { value: fromValue, progress: 0.0 },
+    to: { value: toValue, progress: 1.0 },
+    duration: durationValue,
+    easing: easingValue,
+    step: state => {
+      //var instanceVariable = this.owner.getVariables().get(variable);
+      this._getTween(identifier).progress = state.progress;
+      this._getTween(identifier).currentValue = state.value;
+      //this.owner.setVariableNumber(instanceVariable, state.value) 
+      this.owner.setVariableNumber(variable, state.value);
+      console.log(state);
+    }
+  });
+
+  this._addObjectVariableTween(identifier, newTweenable, variable, fromValue);
+
+  if (destroyObjectWhenFinished) {
+    this._tweens[identifier].instance.tween().then(() => {
+
+      this._removeObjectFromScene(identifier);
+
+    });
+  } else {
+    this._tweens[identifier].instance.tween().then(() => {
+      this._setTweenFinished(identifier, true);
+    });
+  }
+};
+
+/**
+ * Add an object position tween.
+ * @param {string} identifier Unique id to idenfify the tween
+ * @param {number} toX The target X position
+ * @param {number} toY The target Y position
+ * @param {string} easingValue Type of easing
+ * @param {number} durationValue Duration in milliseconds
+ * @param {boolean} destroyObjectWhenFinished Destroy this object when the tween ends
+ */
+gdjs.TweenRuntimeBehavior.prototype.addObjectPositionTween = function (
+  identifier,
+  toX,
+  toY,
+  easingValue,
+  durationValue,
+  destroyObjectWhenFinished
+) {
+
+  if (this._tweenExists(identifier)) {
+    this.removeTween(identifier);
+  }
+
+  var newTweenable = new shifty.Tweenable();
+  newTweenable.setConfig({
+    from: { x: this.owner.getX(), y: this.owner.getY(), progress: 0.0 },
+    to: { x: toX, y: toY, progress: 1.0 },
+    duration: durationValue,
+    easing: easingValue,
+    step: state => {
+      this._getTween(identifier).progress = state.progress;
```

Comment:
Every time you're using this._getTween(identifier) multiple time, could you store it in a local variable to avoid paying multiple time the cost to call the function + the cost of the lookup into the _tween object? (these things are quick, but add up quickly too when you have a lot of objects and tweens :))

```
  var tween = this._getTween(identifier);
  tween.progress = state.progress;
  tween.x = state.x;
  tween.y = state.y;
```

By the way... Are you sure you need to update these values? What is this used for?

Reply:
I intend to add an expression to return the process (and maybe the time that passed) so that other actions can be started. On a paused tween it would even be possible to scrap through the animation.
I think the other properties could indeed be removed.

Reply:
Well on a second thought, I could also add a new option "ping-pong" to the addXXXTween and use the properties to store the start value. Would simplify the creation of moving platforms and the like.

Reply:
How would that ping pong work? I guess this could work by relaunch automatically a tween to go back to the start position when the current tween is finished (and then iteratively, the new tween that is just created is also set as "ping-pong", and when finished will relaunch a new tween to go back at the "start position", which was the previous ending position).

But in this case I think we can still avoid updating these values. The idea is to avoid any work that is not necessary, to make tweens as lightweight as possible. 
Ping pong would be nice, but let's make sure to come up with the cleanest design possible for it - otherwise let's keep this for a later Pull Request/improvement :)

Reply:
Sure, with the ping-pong option the variables would only be set at the start of the animation and swapped at the end. Updating them on each step wouldn't be needed of cause.